client/src/pages/ContestDetails/ContestDetails.tsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { Link } from 'react-router-dom';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import Grid from '@material-ui/core/Grid';
+import Box from '@material-ui/core/Box';
+import Avatar from '@material-ui/core/Avatar';
+import Button from '@material-ui/core/Button';
+import Typography from '@material-ui/core/Typography';
+import ArrowBackIosIcon from '@material-ui/icons/ArrowBackIos';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import FullWidthTabs from './ContestDetailTabs/ContestDetailTabs';
+import { useAuth } from '../../context/useAuthContext';
+import { useSocket } from '../../context/useSocketContext';
+import useStyles from './useStyles';
+import demoProfilePhoto from '../../Images/demo-profile-photo.png';
+import sampleContestData from './SampleContestData';
+
+export default function ContestDetails(): JSX.Element {
+  const classes = useStyles();
+  const { loggedInUser } = useAuth();
+  const { initSocket } = useSocket();
+
+  const history = useHistory();
+  const { title, description, prizeAmount, user, submissions } = sampleContestData();
+
+  useEffect(() => {
+    initSocket();
+  }, [initSocket]);
+
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
+    history.push('/login');
+    // loading for a split seconds until history.push works
+    return <CircularProgress />;
+  }
+
+  return (
+    <Grid container component="main" className={classes.root}>
+      <CssBaseline />
+      <Grid container direction="column">
+        <Grid container alignItems="center" justify="center">
```

Comment:
I think we can make do with just one Grid container instead of two? :)

Reply:
Yup, one suffices! Thanks for checking!